### DISCUSSION TITLE
Customize Happy Face on Display

### DISCUSSION THREAD

**Discussion by LayerCakeMakes** [pvvx/ZigbeeTLc#24] [Ideas]
This is a minor feature request for sure, but it would be nice to be able to customize when the happy/unhappy face is shown.
I, like many others, use the LYWSD03MMC to monitor the dryness of my 3D printer filament, so the lower the better which currently always leads to a unhappy face. I would love to be able to change this.

**Comment by LayerCakeMakes** [comment]
Ahh I think I put this in the wrong repository. Moving this to devbis/z03mmc.

**Comment by pvvx** [comment]
ZHA does not have tools for configuring Zigbee devices.

The answer to your question from devbis is here: 
https://github.com/devbis/z03mmc/issues/61#issuecomment-1807920800

**Reply by LayerCakeMakes** [reply]
Ahh OK I was thinking along the lines of what the firmware supports for "Temperature Display Mode" with celsius/fahrenheit options but for "Smiley Display Mode" with never/comfort/dryer or something along those lines.

The devbis screenshot in the link showing the configuration option in the reporting tab in Z2M would also be an option where to add this, but maybe not quite as convenient.

I understand this is probably not a feature with a huge user base, just thought I'd throw it out there.

**Reply by pvvx** [reply]
Now I add:
![image](https://github.com/pvvx/ZigbeeTLc/assets/12629515/e0035c90-8622-42e1-be76-13cdc1258182)
AttrID 0x0002 - show_smiley - 0 on, 1 - off

**Reply by pvvx** [reply]
Ver 0.1.1.4: According to the "ZigBee Cluster Library Specification" the "Thermostat Temperature Conversion" values have been corrected. Added On/Off smiley. 
ClusterID: 0x0204 (Thermostat User Interface Configuration), Attr: 0x0000 (TemperatureDisplayMode), ENUM8 = 0 in C, = 1 in F, 
![image](https://github.com/pvvx/ZigbeeTLc/assets/12629515/c445a418-9e74-48e4-a8d5-d8dc662a8014)
Attr: 0x0002 (ScheduleProgrammingVisibility), ENUM8 = 0 show smiley On, = 1 show smiley Off
![image](https://github.com/pvvx/ZigbeeTLc/assets/12629515/a453a337-cb63-490b-9813-4cae0786e9f7)

**Reply by LayerCakeMakes** [reply]
That's a nice compromise. Dryer mode (happy below 30% at 20C) might be a bit to special, just disabling the smiley should be good enough.

**Reply by pvvx** [reply]
But this cannot be done in ZHA.

**Reply by LayerCakeMakes** [reply]
OK
